docs/strategy-advanced.md

```diff
@@ -293,20 +283,27 @@ class AwesomeStrategy(IStrategy):
     use_custom_stoploss = True
 
     def custom_stoploss(self, pair: str, trade: 'Trade', current_time: datetime,
-                        current_rate: float, current_profit: float, dataframe: DataFrame,
-                        **kwargs) -> float:
-
+                        current_rate: float, current_profit: float, **kwargs) -> float:
         # Default return value
         result = 1
         if trade:
             # Using current_time directly would only work in backtesting. Live/dry runs need time to
             # be rounded to previous candle to be used as dataframe index. Rounding must also be 
             # applied to `trade.open_date(_utc)` if it is used for `dataframe` indexing.
-            current_time = timeframe_to_prev_date(self.timeframe, current_time)
-            current_row = dataframe.loc[dataframe['date'] == current_time].squeeze()
-            if 'atr' in current_row:
+            dataframe, _ = self.dp.get_analyzed_dataframe(pair, self.timeframe)
+            current_candle = dataframe.loc[-1].squeeze()
+            if 'atr' in current_candle:
                 # new stoploss relative to current_rate
-                new_stoploss = (current_rate - current_row['atr']) / current_rate
+                new_stoploss = (current_rate - current_candle['atr']) / current_rate
+
+                # Round trade date to it's candle time.
+                trade_date = timeframe_to_prev_date(trade.open_date_utc)
+                trade_candle = dataframe.loc[dataframe['date'] == trade_date]
+                # Just opened trades do not have their candle complete yet therefore trade_candle may be None
+                if trade_candle is not None:
+                    trade_candle = trade_candle.squeeze()
+                    trade_stoploss = (current_rate - trade_candle['atr']) / current_rate
+                    new_stoploss = max(new_stoploss, trade_stoploss)
```

Comment:
i guess this part is now outdated?

Reply:
Uhm... Not exactly? I mean i updated such example in other spot in https://github.com/freqtrade/freqtrade/pull/4851/commits/17b9e898d2a234b9c1861192af21618ca63442a1 to showcase getting both last candle and trade candle. This bit here is not incorrect, but maybe incomplete as it does not show getting last candle.

Reply:
ah you're right - it's to get the candle at the trade open - not the latest candle - i misread that ... All fine then 👍